### PR TITLE
Fix #4200 Add dark mode support to AudioLanguageActivity, ReadingTextSizeActivity, AppLanguageActivity and dialog views.

### LINKED ISSUE #4200: Add dark mode support to AudioLanguageActivity, ReadingTextSizeActivity, AppLanguageActivity and dialog views.
Add dark mode support to AudioLanguageActivity, ReadingTextSizeActivity, AppLanguageActivity and dialog views.
Dialog views include:
- profile_picture_edit_dialog
- cellular_data_dialog 
- etc.
and the system AlertDialog. (look at OppiaAlertDialogTheme)

Go through this [wiki](https://github.com/oppia/oppia-android/wiki/Dark-Mode) to get started.

### PR DESCRIPTION
**Pull request by indresh149**
## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by indresh149** [comment]
@ayush0402 can you please review this ?

File: app/build.gradle [hunk 1]
```diff
@@ -6,12 +6,12 @@ apply plugin: 'kotlin-android-extensions'
...
 
 android {
-  compileSdkVersion 29
+  compileSdkVersion 31
   buildToolsVersion "29.0.2"
   defaultConfig {
     applicationId "org.oppia.android"
     minSdkVersion 19
-    targetSdkVersion 29
+    targetSdkVersion 31
```
Comment (ayush0402) [inline app/build.gradle:14]: Revert these changes.

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -88,19 +88,19 @@
     <activity
       android:name=".app.options.AppLanguageActivity"
       android:label="@string/app_language_activity_title"
-      android:theme="@style/OppiaThemeWithoutActionBar" />
+      android:theme="@style/AppTheme.darkMode" />
     <activity
       android:name=".app.options.AudioLanguageActivity"
       android:label="@string/audio_language_activity_title"
-      android:theme="@style/OppiaThemeWithoutActionBar" />
+      android:theme="@style/AppTheme.darkMode" />
```
Comment (ayush0402) [inline app/src/main/AndroidManifest.xml:95]: We are not aiming to have newer themes here for dark mode. The activities should follow default theme only but the color are needed to be separated for the variations. The changes should only happen in the color files and the concerned layout files unless some special cases. Please go through the wiki for it again and ask any questions if its still not clear.

File: app/src/main/java/org/oppia/android/app/help/faq/faqsingle/FAQSingleActivity.kt [hunk 3]
```diff
@@ -18,7 +18,11 @@ class FAQSingleActivity : InjectableAppCompatActivity() {
     (activityComponent as ActivityComponentImpl).inject(this)
     val question = intent.getStringExtra(FAQ_SINGLE_ACTIVITY_QUESTION)
     val answer = intent.getStringExtra(FAQ_SINGLE_ACTIVITY_ANSWER)
-    faqSingleActivityPresenter.handleOnCreate(question, answer)
+    if (question != null) {
+      if (answer != null) {
+        faqSingleActivityPresenter.handleOnCreate(question, answer)
+      }
+    }
```
Comment (ayush0402) [inline app/src/main/java/org/oppia/android/app/help/faq/faqsingle/FAQSingleActivity.kt:25]: Why are these changes needed?

File: app/src/main/java/org/oppia/android/app/player/exploration/ExplorationActivity.kt [hunk 4]
```diff
@@ -54,9 +54,9 @@ class ExplorationActivity :
     super.onCreate(savedInstanceState)
     (activityComponent as ActivityComponentImpl).inject(this)
     internalProfileId = intent.getIntExtra(EXPLORATION_ACTIVITY_PROFILE_ID_ARGUMENT_KEY, -1)
-    topicId = intent.getStringExtra(EXPLORATION_ACTIVITY_TOPIC_ID_ARGUMENT_KEY)
-    storyId = intent.getStringExtra(EXPLORATION_ACTIVITY_STORY_ID_ARGUMENT_KEY)
-    explorationId = intent.getStringExtra(EXPLORATION_ACTIVITY_EXPLORATION_ID_ARGUMENT_KEY)
+    topicId = intent.getStringExtra(EXPLORATION_ACTIVITY_TOPIC_ID_ARGUMENT_KEY).toString()
+    storyId = intent.getStringExtra(EXPLORATION_ACTIVITY_STORY_ID_ARGUMENT_KEY).toString()
+    explorationId = intent.getStringExtra(EXPLORATION_ACTIVITY_EXPLORATION_ID_ARGUMENT_KEY).toString()
```
Comment (ayush0402) [inline app/src/main/java/org/oppia/android/app/player/exploration/ExplorationActivity.kt:59]: Ditto. This should not be a part of this PR.

File: app/src/main/java/org/oppia/android/app/profile/AdminAuthActivityPresenter.kt [hunk 5]
```diff
@@ -92,10 +92,12 @@ class AdminAuthActivityPresenter @Inject constructor(
...
-      } else if (inputPin.length == adminPin.length) {
-        authViewModel.errorMessage.set(
-          resourceHandler.getStringInLocale(R.string.admin_auth_incorrect)
-        )
+      } else if (adminPin != null) {
+        if (inputPin.length == adminPin.length) {
+          authViewModel.errorMessage.set(
+            resourceHandler.getStringInLocale(R.string.admin_auth_incorrect)
+          )
+        }
```
Comment (ayush0402) [inline app/src/main/java/org/oppia/android/app/profile/AdminAuthActivityPresenter.kt:100]: Ditto.

File: app/src/main/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityPresenter.kt [hunk 6]
```diff
@@ -92,56 +92,54 @@ class QuestionPlayerActivityPresenter @Inject constructor(
...
+    startDataProvider?.toLiveData()?.observe(
+      activity
+    ) {
+      when {
+        it.isPending() -> oppiaLogger.d(
+          "QuestionPlayerActivity",
+          "Starting training session"
+        )
+        it.isFailure() -> {
+          oppiaLogger.e(
```
Comment (ayush0402) [inline app/src/main/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityPresenter.kt:105]: Ditto for this and other similar changes.

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -312,7 +312,7 @@
   <string name="profile_edit_allow_download_heading">Allow Download Access</string>
   <string name="profile_edit_allow_download_sub">User is able to download and delete content without Administrator password</string>
   <!-- ProfilePictureActivity -->
-  <string name="profile_picture_activity_title">Profile Picture</string>
+  <string name="profile_picture_activity_title" translatable="false">Profile Picture</string>
```
Comment (ayush0402) [inline app/src/main/res/values/strings.xml:315]: Again, not a part of this PR. Ditto at other places.

**Review by ayush0402 — CHANGES_REQUESTED** [review]
@indresh149 Sorry for the delayed review but this PR has too many unwanted changes. Please go through dark mode wiki again and let me know if something is not clear. Thanks.

File: app/build.gradle (hunk 1 above)
Comment (indresh149) [reply app/build.gradle:14]: okk sure

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (indresh149) [reply app/src/main/AndroidManifest.xml:95]: okk , understood , i will try to do it by this way now.

File: app/src/main/java/org/oppia/android/app/help/faq/faqsingle/FAQSingleActivity.kt (hunk 3 above)
Comment (indresh149) [reply app/src/main/java/org/oppia/android/app/help/faq/faqsingle/FAQSingleActivity.kt:25]: there is some issue in the Gradle build, android studio suggested changing these.

File: app/src/main/java/org/oppia/android/app/player/exploration/ExplorationActivity.kt (hunk 4 above)
Comment (indresh149) [reply app/src/main/java/org/oppia/android/app/player/exploration/ExplorationActivity.kt:59]: okk

File: app/src/main/java/org/oppia/android/app/profile/AdminAuthActivityPresenter.kt (hunk 5 above)
Comment (indresh149) [reply app/src/main/java/org/oppia/android/app/profile/AdminAuthActivityPresenter.kt:100]: okk

File: app/src/main/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityPresenter.kt (hunk 6 above)
Comment (indresh149) [reply app/src/main/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityPresenter.kt:105]: okk

File: app/src/main/res/values/strings.xml (hunk 7 above)
Comment (indresh149) [reply app/src/main/res/values/strings.xml:315]: okk

**Comment by rt4914** [comment]
Un-assigning until approved / re-assigned by @ayush0402
